src/openrct2/peep/peep.c

```diff
@@ -5565,6 +5565,31 @@ static void peep_update_walking_break_scenery(rct_peep* peep){
     sint32 edges = map_element->properties.path.edges & 0xF;
     if (edges == 0xF) return;
 
+    uint16 sprite_id = sprite_get_first_in_quadrant(peep->x, peep->y);
+
+    // Check if a peep is already sitting on the bench.
```

Comment:
`// Check if a peep is already sitting on the bench. If so, do not vandalise it.`